src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -575,6 +574,9 @@ bool D4NFilterObject::check_head_exists_in_cache_get_oid(const DoutPrefixProvide
       found_in_cache = false;
       ldpp_dout(dpp, 10) << "D4NFilterObject::" << __func__ << "(): CacheDriver get_attrs method failed." << dendl;
     }
+    
+    // get_attrs overwrite version, so set below
+    this->set_object_version(version);
```

Comment:
Can you explain how get_attrs() overwrites the version? and which get_attrs()? cachedriver->get_attrs() does not deal with the object version at all

Reply:
Removing this change because it's unnecessary.